src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets

```diff
@@ -129,6 +129,15 @@
       <SourcesToCopyToTestDir Include="@(OverridenRuntimeContent)" />
       <DestinationsForTestDir Include="@(OverridenRuntimeContent->'$(TestPath)\%(Filename)%(Extension)')" />
     </ItemGroup>
+    
+    <!-- Removing assets from TestCopyLocal that come in from P2P references in SourcesToCopyToTestDir -->
+    <ItemGroup>
+        <_FileNamesInSourcesToCopyToTestDir Include="@(SourcesToCopyToTestDir->'%(Filename)%(Extension)')">
+          <SourcePath>%(SourcesToCopyToTestDir.Identity)</SourcePath>
+        </_FileNamesInSourcesToCopyToTestDir>
+        <_DeduppedListOfTestAssetsByFileName Include="@(_TestCopyLocalByFileNameWithoutDuplicates)" Exclude="@(_FileNamesInSourcesToCopyToTestDir)" />
```

Comment:
nit: to me a more useful name might be "finalTestAssetsByFileName".  Something to make it clear that this is the list that folks ought to rely on for finding all the final test assemblies.  Who knows if that will actually prevent folks from further forking this in the future... but at least its an attempt.